src/test/java/uk/nhs/prm/deduction/e2e/tests/RepositoryE2ETests.java

```diff
@@ -280,6 +278,131 @@ void shouldVerifyThatALargeEhrXMLIsUnchanged() {
+        );
+    }
+
+    @ParameterizedTest(name = "[{index}] Case of {0}")
+    @MethodSource("erroneousInboundMessages")
+    @DisplayName("Test how ORC handles Erroneous inbound messages")
+    void testsWithErroneousInboundMessages(String inboundMessage, String conversationId) {
+        // when
+        inboundQueueFromMhs.sendMessage(inboundMessage, conversationId);
+
+        // then
+        // verify that EHR-IN put the message to unhandled queue
+        SqsMessage unhandledMessage = ehrInUnhandledQueue.getMessageContaining(conversationId);
+        assertThat(unhandledMessage.body()).isEqualTo(inboundMessage);
+
+        // verify that no response message with given conversation id is on the gp2gp observability queue
+        // later this could be changed to asserting an NACK message on the queue if we do send back NACKs
+        assertThat(gp2gpMessengerQueue.verifyNoMessageContaining(conversationId)).isTrue();
+
+        checkThatAllServicesHealthCheckPassing();
+    }
+
+    private void checkThatAllServicesHealthCheckPassing() {
+        List<String> servicesToCheck = List.of(
+                config.getEhrRepoUrl(),
+                config.getEhrOutUrl(),
+                config.getGp2GpMessengerUrl()
+        );
+        boolean allHealthChecksPassing = servicesToCheck.stream().allMatch(HealthCheck::isHealthCheckPassing);
+        assertThat(allHealthChecksPassing).isTrue();
+    }
+
+
+
+    private void addSmallEhrToEhrRepo(String nhsNumber) {
+        // A utility function to add a small ehr of given patient into ehr REPO
+
+        String inboundConversationId = createUuidAsString();
+        String smallEhrMessageId = createUuidAsString();
+        String previousGpForTestPatient = "M85019";
+
+        // TODO: Consider to implement a better way of filling details into the EHR files
+        String smallEhr = getSmallEhrWithoutLinebreaks(inboundConversationId, smallEhrMessageId)
+                .replaceAll("9727018440", nhsNumber);
+
+        addRecordToTrackerDb(trackerDb, inboundConversationId, "", nhsNumber, previousGpForTestPatient, "ACTION:EHR_REQUEST_SENT");
+        inboundQueueFromMhs.sendMessage(smallEhr, inboundConversationId);
+
+        trackerDb.waitForStatusMatching(inboundConversationId, "ACTION:EHR_TRANSFER_TO_REPO_COMPLETE");
+    }
+
+    @Test
+    void shouldRejectDuplicatedEhrRequestMessagesFromTheSameGP() {
+        // given
+        Patient testPatient = Patient.PATIENT_WITH_SMALL_EHR_RECORD_IN_REPO_AND_MOF_SET_TO_TPP;
+        String nhsNumberForTest = testPatient.nhsNumber();
+
+        addSmallEhrToEhrRepo(nhsNumberForTest);
+
+        EhrRequestMessage ehrRequestMessage = new EhrRequestMessageBuilder()
+                .withNhsNumber(nhsNumberForTest)
+                .withEhrSourceAsRepo(config)
+                .withEhrDestinationGp(Gp2GpSystem.TPP_PTL_INT)
+                .build();
+        int numberOfDuplicatedMessage = 3;
```

Comment:
Consider pluralising this variable to 'numberOfDuplicatedMessages'
```suggestion
        int numberOfDuplicatedMessages = 3;
```